d2l-gestures-swipe.html

```diff
@@ -0,0 +1,105 @@
+<script>
+(function() {
+	'use strict';
+
+	var Swipe = {
+
+		_maxTime: 2000,
+
+		_handleTouchStart: function(e) {
+
+			var node = this;
+
+			var tracking = {
+				start: {
+					time: new Date().getTime(),
+					x: e.touches[0].clientX,
+					y: e.touches[0].clientY
+				}
+			};
+
+			var reset = function() {
+				tracking = null;
+				node.removeEventListener('touchend', handleTouchEnd);
+				node.removeEventListener('touchermove', handleTouchMove);
+				node.removeEventListener('touchcancel', handleTouchCancel);
+			};
+
+			var handleTouchCancel = function() {
```

Comment:
`touchcancel` can be dispatched by the browser for a variety of reasons (ex. too many touch points on the device), and we need to handle these cases by removing event listeners, etc..